src/transformers/modeling_utils.py

```diff
@@ -2962,6 +3013,13 @@ def from_pretrained(
                 kwargs["skip_keys"] = model._skip_keys_device_placement
             dispatch_model(model, **kwargs)
 
+        if quantization_method_from_args == QuantizationMethod.GPTQ:
+            from . import AutoTokenizer
+
+            tokenizer = AutoTokenizer.from_pretrained(pretrained_model_name_or_path, use_fast=True)
```

Comment:
It is really weird to me that the quantizer requires both the model and the tokenizer. This creates a dependency on modeling_utils toward the tokenization files which is not warranted nad might create error for users if the checkpoint for tokenization is different from the checkpoint for the model.

Is there a way to avoid this?

Reply:
Yes, the quantization process needs to run inference on some data. I moved the tokenizer in the quantization config. After the quantization is done, we don't need it anymore so there is no issue if we can't save it.

Reply:
But we have vision models, speech models and multimodal models :-/